packages/relay/src/lib/eth.ts

```diff
@@ -1211,11 +1211,23 @@ export class EthImpl implements Eth {
       ]);
       if (result) {
         if (result?.type === constants.TYPE_TOKEN) {
+          const blockInfo = await this.getBlockInfo(blockNumber, requestDetails);
+          if (blockInfo) {
+            if (parseFloat(result.entity?.created_timestamp) > parseFloat(blockInfo.timestamp.to)) {
```

Comment:
Mmm, I'm not convinced about the timestamp comparison. I guess the issue is with `resolveEntityType`, which should get the `blockNumber`/`timestamp`. I'm not sure if by doing the comparison here we are missing any edge case. If it gets too complicated, then we can leave like this and think of a refactor in the future. WDYT?

Reply:
Generally I agree and it also makes the method more confusing, however I did try just passing the timestamp to `resolveEntityType`, but at the end of the try block there is this fallback logic which gets the contract bytecode from the SDK as fallback in case the mirror node returns empty bytecode or 404. 

```
      const bytecode = await this.hapiService
        .getSDKClient()
        .getContractByteCode(0, 0, address, EthImpl.ethGetCode, requestDetails);
      return prepend0x(Buffer.from(bytecode).toString('hex'));
```

However if I pass the timestamp there might be a case where the fallback logic gets triggered for an earlier block and this would be harder to check.  I would suggest to leave it like this and possibly refactor/simplify the whole method

Reply:
Sounds good to me.